lib/msal-common/src/request/ScopeSet.ts

```diff
@@ -21,29 +21,45 @@ export class ScopeSet {
     // Boolean denoting whether scopes are required. Usually used for validation.
     private scopesRequired: boolean;
 
-    constructor(inputScopes: Array<string>, appClientId: string, scopesRequired: boolean) {
-        this.clientId = appClientId;
+    constructor(
+        inputScopes: Array<string>,
+        clientId: string,
+        scopesRequired: boolean,
+        nonSPA?: boolean
+    ) {
+        this.clientId = clientId;
         this.scopesRequired = scopesRequired;
+
         // Filter empty string and null/undefined array items
-        const filteredInput = inputScopes ? StringUtils.removeEmptyStringsFromArray(inputScopes) : inputScopes;
+        const filteredInput = (inputScopes)
+            ? StringUtils.removeEmptyStringsFromArray(inputScopes)
+            : inputScopes;
+
         // Validate and filter scopes (validate function throws if validation fails)
         this.validateInputScopes(filteredInput);
-        const scopeArr = filteredInput ? StringUtils.trimAndConvertArrayEntriesToLowerCase([...filteredInput]) : [];
+
+        const scopeArr = filteredInput? StringUtils.trimAndConvertArrayEntriesToLowerCase([...filteredInput]): [];
         this.scopes = new Set<string>(scopeArr);
         if (!this.scopesRequired) {
             this.appendScope(this.clientId);
         }
         this.originalScopes = new Set<string>(this.scopes);
-        this.replaceDefaultScopes();
+
+        // add default scopes
+        nonSPA ? this.addDefaultScopes() : this.replaceDefaultScopes();
```

Comment:
I don't think this is needed. We should revisit the changes in this class.

Reply:
I've updated to get rid of nonSpa. Ultimately I think scopesRequired should be refactored as well, passing in boolean as parameter is a code smell, but I'll leave that for a separate PR.